invenio_pages/views.py

```diff
@@ -30,7 +30,8 @@ def register_pages():
     current_app.view_functions["invenio_pages.view"] = view
 
     for page in Page.query.all():
-        _add_url_rule(page.url)
+        if not page.has_custom_view:  # Check if page does not have a custom view
```

Comment:
nit: I would add a small comment on why this was added. I got context from @yashlamba